tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -1,51 +1,135 @@
+
+Test Duplicate Image
+    [Documentation]  Test adding two images with the same name (should fail)
+    Apply Custom ImageStream And Check Status
+    Sleep  1
+    Import New Image    ${IMG_URL}    ${IMG_NAME}    ${IMG_DESCRIPTION}
+    ...    software=${IMG_SOFTWARE}
+    ...    packages=${IMG_PACKAGES}
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
+
+Test Bad Image URL
+    [Documentation]  Test adding an image with a bad repo URL (should fail)
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  quay.io/RandomName/RandomImage:v1.2.3
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    ${IMG_URL}=  Set Variable  ${OG_URL}
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Reset Image Name
+    # No image created, teardown not needed
+
+Test Bad Image Import
+    [Documentation]  Import a broken image and confirm it cannot spawn
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  randomstring
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
+    Get ImageStream Metadata And Check Name
     Launch JupyterHub Spawner From Dashboard
-    Spawn Notebook With Arguments  image=${IMG_NAME}  size=Default
+    Run Keyword And Continue On Failure  Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Default
+    ${IMG_URL}=  Set Variable  ${OG_URL}
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Reset Image Name
 
 
 *** Keywords ***
 Custom Image Teardown
     [Documentation]    Closes the JL server and deletes the ImageStream
-    End Web Test
-    OpenShiftCLI.Delete    kind=ImageStream    field_selector=metadata.name==${IMG_NAME}
-    ...    namespace=redhat-ods-applications
+    Clean Up Server
+    Stop JupyterLab Notebook Server
+    Go To  ${ODH_DASHBOARD_URL}
+    Open Notebook Images Page
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
 
 Apply Custom ImageStream And Check Status
-    [Documentation]    Applies a custom ImageStream as a YAML and checks the status
-    ${apply_result} =    Run Keyword And Return Status    Run    oc apply -f ${YAML}
-    Should Be Equal    "${apply_result}"    "True"
-    # OpenShiftCLI.Apply    kind=ImageStream    src="tests/Resources/Files/custom_image.yaml"
-    # ...    namespace=redhat-ods-applications
+    [Documentation]    Imports a custom ImageStream via UI and checks the status
+    ${curr_date} =  Get Time  year month day hour min sec
+    ${curr_date} =  Catenate  SEPARATOR=  @{curr_date}
+
+    # Create a unique notebook name for this test run
+    ${IMG_NAME} =  Catenate  ${IMG_NAME}  ${curr_date}
+    Set Global Variable  ${IMG_NAME}  ${IMG_NAME}
+
+    ${apply_result} =    Run Keyword And Return Status    Import New Image
```

Comment:
should this variable be checked? asking since the kw title says `Check Status`

Reply:
Mh, I'll probably just rename this since we already check that the image is correctly displayed in the TC itself